test/models/weather_request_test.rb

```diff
@@ -34,4 +36,40 @@ def setup
       @weather_request.destroy
     end
   end
+  
+  test "should extract weather info" do
+    expect_info = OpenStruct.new(weather_time: 'Wed, 24 Oct 2018 06:00 PM AEDT', temperature: 17, feels_like: 16, description: 'Cloudy', sunrise: '6:4 am', sunset: '7:17 pm')
+    Net::HTTP.expects(:get).returns(@json_valid)
+    weather_request = WeatherRequestService.new(@location).run
+    weather_info = weather_request.weather_info
+    assert_equal weather_info.weather_time, expect_info.weather_time
+    assert_equal weather_info.temperature,  expect_info.temperature
+    assert_equal weather_info.feels_like,   expect_info.feels_like
+    assert_equal weather_info.description,  expect_info.description
+    assert_equal weather_info.sunrise,      expect_info.sunrise
+    assert_equal weather_info.sunset,       expect_info.sunset
+  end
+  
+  test "should extract weather forecast" do
+    expect_forecasts = [OpenStruct.new(date: '24 Oct 2018', day_name: 'Wed', temp_high: 18, temp_low: 15, description: 'Mostly Cloudy'),
```

Comment:
Could this be done with a fixture or a factory?

Reply:
Not sure about a fixture, but probably a factory.  If I had more time I'd switch the OpenStructs to models and then probably use fixtures.  I'm not sure how to use factories with minitest, but honestly I didn't think to check.

Reply:
That's all cool.  Just discussing options.

I was thinking you could use a CSV and create `OpenStruct`s as you read each line.  That would be easier to add or modify data.